### DISCUSSION TITLE
Best way to add NextJS to existing project

### DISCUSSION THREAD

**Discussion by S7012MY** [vercel/next.js#34318] [Help] [answered]
Currently I'm pulling some data from an api, but I want to be able to manipulate that data through a web app.
I tried running create-next-app inside the project folder, but I end up with a sub folder containing a separate package.json and I'm not sure that's the proper way to do it.
What's the best way to do this?
Thanks!

**Comment by GabenGar** [comment]
Follow the manual setup section:
https://nextjs.org/docs/getting-started#manual-setup

**Comment by 2stash** [comment, marked as answer]
If @GabenGar didn't answer your question then you need to elaborate where you are currently at in your development and what you have on the frontend. If you have a React app then you just upgrade per the docs in @GabenGar 's answer. Otherwise if you have something else for a frontend then you need to probably start over with create-next-app and migrate some code. Or if what you are describing is the backend, then you can have the nextjs frontend in a different folder with it's own package.json, which is standard for frontend/backend applications. Please accept @GabenGar 's answer or elaborate.

Good luck!

**Reply by S7012MY** [reply]
Thanks a lot for the answers. It's a good starting point. I don't have a front end or a back end, just a crawler populating a database and a prisma schema. I like create-next-app because it generates all the folders and I have a good starting point (I don't have much experience with anything else besides react in js).

**Reply by 2stash** [reply]
Yeah then just leave your crawler in a separate folder/repo, and the start a new nextjs in another folder, then you won't have conflicting package.json. Nextjs can handle the backed for the calls to your database as well.
